### PR TITLE
feat: introduce `modules` property to flix.toml

### PR DESCRIPTION
**Pull request by sockmaster27**
Related to #6377

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/tools/pkg/IncludedModules.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.tools.pkg
+
+import ca.uwaterloo.flix.language.ast.Symbol
+
+sealed trait IncludedModules {
+  def contains(sym: Symbol.ModuleSym): Boolean
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/tools/pkg/IncludedModules.scala:21]: The implementation should be here, by pattern matching on this.

We dont use the OO style with override :)

File: main/src/ca/uwaterloo/flix/tools/pkg/IncludedModules.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.tools.pkg
+
+import ca.uwaterloo.flix.language.ast.Symbol
+
+sealed trait IncludedModules {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/tools/pkg/IncludedModules.scala:20]: @mlutze Can we come up with a better name? Perhaps just PackageModules?

File: main/src/ca/uwaterloo/flix/tools/pkg/ManifestParser.scala [hunk 3]
```diff
@@ -469,6 +487,31 @@ object ManifestParser {
...
+  /**
+    * Creates the `IncludedModules` object from `optArray`.
+    */
+  private def toIncludedModules(optArray: Option[TomlArray], p: Path): Result[IncludedModules, ManifestError] = {
+    optArray match {
+      case None =>
+        Ok(IncludedModules.All)
+      case Some(array) =>
+        var moduleSet = Set.empty[Symbol.ModuleSym]
+        for (i <- 0 until array.size()) {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/tools/pkg/ManifestParser.scala:499]: Is it not possible to iterate directly through the array?

**Comment by magnus-madsen** [comment]
Ill let @mlutze have a look too.

File: main/src/ca/uwaterloo/flix/tools/pkg/IncludedModules.scala (hunk 2 above)
Comment (mlutze) [reply main/src/ca/uwaterloo/flix/tools/pkg/IncludedModules.scala:20]: ModuleSet?

File: main/src/ca/uwaterloo/flix/tools/pkg/ManifestParser.scala [hunk 4]
```diff
@@ -469,6 +487,31 @@ object ManifestParser {
...
+          try {
+            val string = array.getString(i)
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/tools/pkg/ManifestParser.scala:501]: What happens if the array contains a non-string? Maybe we should do this validation in the parsing part?

File: main/src/ca/uwaterloo/flix/tools/pkg/IncludedModules.scala (hunk 2 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/tools/pkg/IncludedModules.scala:20]: Let's call it `PackageModules`. That makes the most sense to me.

File: main/src/ca/uwaterloo/flix/tools/pkg/Manifest.scala [hunk 5]
```diff
@@ -21,6 +21,7 @@ case class Manifest(name: String,
                     description: String,
                     version: SemVer,
                     repository: Option[GitHub.Project],
+                    modules: IncludedModules,
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/tools/pkg/Manifest.scala:24]: this field should still be called just `modules` that is fine.

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**